Profile and Register Page

mertzmel opened the pull request:
Refactored the register page this morning to use the Controller and the react-hook-form, all the error cases worked (gave the right message for blank fields, bad Rnumber, password mismatch, and password being too weak).
When all the register info is the correct format, the register button won't do anything but clear most of the fields (the fields also clear if there are problems with the info, but that's a little more minor). It doesn't do make post request and I can't figure out why.
The profile page is set up with the new refactored template we made from the register page. We also added a little link to get back to the homepage. We have theories on how we would fill in the information, but we can't figure out how to pull from the database for user information.

Akhil-Kapadia committed:
Merge pull request #6 from Akhil-Kapadia/frontend--Routing-Rework

V : [0:2:0] - rerouting rework

mertzmel committed:
Register and Profile Pages

- Refactored the register page this morning to use the Controller and the react-hook-form, all the error cases worked (gave the right message for blank fields, bad Rnumber, password mismatch, and password being too weak). When all the register info is the correct format, the register button won't do anything but clear most of the fields (the fields also clear if there are problems with the info, but that's a little more minor). It doesn't do make post request and I can't figure out why
- The profile page is set up with the new refactored template we made from the register page. We also added a little link to get back to the homepage. We have theories on how we would fill in the information, but we can't figure out how to pull from the database for user information.

Akhil-Kapadia reviewed (changes requested):
- **frontend/package-lock.json & package.json** is crazy big. IDK what happened but its 17k lines and can't even be viewed online without downloading.
-  **App.js** : The profile route is in the incorrect place, it should go under /profile/user which is line 85.
- **pointsForm.js** : Discard changes to this file. (Being reworked in "propoint - moderation" branch) 
- **pointsTable.js**  : Discard changes to this file. (Being reworked in "propoint - moderation" branch)
- **auth.js**: Discard changes to this file. The backend needs to be refactored and that will be done in a different branch. Also see below.
- **register.js / profile.js** : Overall good work, One change, to reduce overhead on the server the validation will be done on the client side before sending a valid request to the backend. This prevents unnecessary resets and page reloads. 
So, what basically needs to be done is using the react hook forms in-built validator. Reference line 109 in login.js and visit https://www.react-hook-form.com/api/useform/register for props for the rules={{stuff}}. To validate two different fields you will have to use a callback function for validation. Check the docs for react - hook - forms to see how to do that.

Thanks and message me any questions, as it is I can't merge into the main branch yet.

mertzmel committed:
Profile & Register part 2

frontend/package-lock.json & package.json - left alone
App.js : moved /profile/user to correct line
pointsForm.js : Discarded changes to this file
pointsTable.js : Discarded changes to this file
auth.js: Discarded changes to this file
register.js / profile.js : added validation in register.js file

Akhil-Kapadia commented:
Implemented
